test/common/http/conn_manager_impl_test_base.h

```diff
@@ -212,6 +213,32 @@ class HttpConnectionManagerImplTest : public testing::Test, public ConnectionMan
   std::vector<MockStreamDecoderFilter*> decoder_filters_;
   std::vector<MockStreamEncoderFilter*> encoder_filters_;
   std::shared_ptr<AccessLog::MockInstance> log_handler_;
+
+  // For testing Router::DelegatingRoute
+  class ExampleDelegatingRouteEntryDerived : public Router::DelegatingRouteEntry {
+  public:
+    ExampleDelegatingRouteEntryDerived(const Router::RouteEntry* base_route_entry,
+                                       const std::string& cluster_name_override)
+        : DelegatingRouteEntry(base_route_entry), custom_cluster_name_(cluster_name_override) {}
+
+    const std::string& clusterName() const override { return custom_cluster_name_; }
+
+  private:
+    const std::string& custom_cluster_name_;
+  };
+
+  class ExampleDelegatingRouteDerived : public Router::DelegatingRoute {
+  public:
+    ExampleDelegatingRouteDerived(Router::RouteConstSharedPtr base_route,
+                                  const std::string& cluster_name_override)
+        : DelegatingRoute(base_route), custom_route_entry_(new ExampleDelegatingRouteEntryDerived(
```

Comment:
make `custom_route_entry` a unique_ptr and do `std::make_unique< ExampleDelegatingRouteEntryDerived>(base_route->routeEntry(), cluster_name_override)` to avoid a memory leak